From axel.lin@gmail.com  Thu Aug 19 10:26:44 2010
Subject: USB: r8a66597-udc: return -ENOMEM if kzalloc() fails
From: Axel Lin <axel.lin@gmail.com>
Cc: Yoshihiro Shimoda <yoshihiro.shimoda.uh@renesas.com>,
	Greg Kroah-Hartman <gregkh@suse.de>,
	Paul Mundt <lethal@linux-sh.org>, linux-usb@vger.kernel.org
Date: Tue, 17 Aug 2010 09:41:29 +0800
Message-Id: <1282009289.18724.0.camel@mola>

Signed-off-by: Axel Lin <axel.lin@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/gadget/r8a66597-udc.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/usb/gadget/r8a66597-udc.c
+++ b/drivers/usb/gadget/r8a66597-udc.c
@@ -1557,6 +1557,7 @@ static int __init r8a66597_probe(struct
 	/* initialize ucd */
 	r8a66597 = kzalloc(sizeof(struct r8a66597), GFP_KERNEL);
 	if (r8a66597 == NULL) {
+		ret = -ENOMEM;
 		printk(KERN_ERR "kzalloc error\n");
 		goto clean_up;
 	}
